extensions/mssql/src/baseService.ts

```diff
@@ -0,0 +1,30 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { RequestType } from 'vscode-languageclient';
+import { SqlOpsDataClient } from 'dataprotocol-client';
+
+/**
+ * Base class containing shared code to reduce boilerplate for services
+ */
+export abstract class BaseService {
+	constructor(protected readonly client: SqlOpsDataClient) { }
+
+	/**
+	 * Runs the specified request wrapped in the requisite try-catch
+	 * @param type RequestType, typically in the format 'contracts.DoThingRequest.type'
+	 * @param params parameters to be passed to the request
+	 * @returns result from the request
+	 */
+	protected async runWithErrorHandling<P, R, E, RO>(type: RequestType<P, R, E, RO>, params: P): Promise<R> {
```

Comment:
I am sure others would appreciate if you could update all the services :)

Reply:
That's the plan!  Just wanted to start with the services I knew best in case y'all spotted any issues.

Reply:
I think we can probably define in the dataprotocol-client so that even other extensions can use this.

Reply:
+1 on that, would love to see that added there

Reply:
Good call!  I'll add it there, and swap this over once the new version goes out.